mapping/mapping_with_vector.go

```diff
@@ -0,0 +1,129 @@
+//  Copyright (c) 2023 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build vectors
+// +build vectors
+
+package mapping
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/blevesearch/bleve/v2/document"
+	"github.com/blevesearch/bleve/v2/util"
+)
+
+// Default Similarity value for vector fields
+const SimilarityDefaultVal = "l2_norm"
+
+// Set of Valid Similarity values for vector fields
+var SimilarityValidVals = map[string]struct{}{
+	"l2_norm":     {},
+	"cosine":      {},
+	"dot_product": {},
```

Comment:
Given that we're normalizing vector data - I don't see the need for cosine & inner dot product - they'd be the same yes.

Reply:
@abhinavdangeti
I think we are not normalizing vectors submitted by the user.
( Although, we can create an analyzer in bleve, which users can choose, if they want to normalize their vectors before indexing/querying them)

Regardless,
"cosine" is a special case of "dot_product",
so, we can keep "dot_product" and let user know that, if they want "cosine" similarity, they can normalize their vectors either at application layer, or at analysis pipe line.